Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp

```diff
@@ -0,0 +1,537 @@
+/*
+ * Copyright (C) 2018, 2019, 2021, 2024 Igalia S.L
+ * Copyright (C) 2018, 2019 Zodiac Inflight Innovations
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "WPEQtView.h"
+
+#include "WPEViewQtQuick.h"
+#include "WPEQtViewLoadRequest.h"
+#include "WPEQtViewLoadRequestPrivate.h"
+
+#include <QQuickWindow>
+#include <QSGSimpleTextureNode>
+#include <wtf/glib/GUniquePtr.h>
+
+/*!
+  \qmltype WPEView
+  \inqmlmodule org.wpewebkit.qtwpe
+  \brief A component for displaying web content.
+
+  WPEView is a component for displaying web content which is implemented using native
+  APIs on the platforms where this is available, thus it does not necessarily require
+  including a full web browser stack as part of the application.
+
+  WPEView provides an API compatible with Qt's QtWebView component. However
+  WPEView is limited to Linux platforms supporting EGL KHR extensions. WPEView
+  was successfully tested with the EGLFS and Wayland-EGL QPAs.
+*/
+WPEQtView::WPEQtView(QQuickItem* parent)
+    : QQuickItem(parent)
+{
+    connect(this, &QQuickItem::windowChanged, this, &WPEQtView::configureWindow);
+    setFlag(ItemHasContents, true);
+    setAcceptedMouseButtons(Qt::AllButtons);
+    setAcceptHoverEvents(true);
+    setAcceptTouchEvents(true);
+}
+
+WPEQtView::~WPEQtView()
+{
+    if (m_webView)
+        g_signal_handlers_disconnect_by_data(m_webView.get(), this);
+}
+
+void WPEQtView::geometryChange(const QRectF& newGeometry, const QRectF&)
+{
+    m_size = newGeometry.size().toSize();
+    if (!m_webView)
+        return;
+
+    if (auto* wpeView = webkit_web_view_get_wpe_view(m_webView.get()))
+        wpe_view_resize(wpeView, m_size.width(), m_size.height());
```

Comment:
Is this expected to work with the old api too? Because when using the new api `webkit_web_view_get_wpe_view` should always return a non-null pointer.